File: python/pangram/pangram.py
import string

def is_pangram(sentence):
    if not isinstance(sentence, str):
        raise TypeError('argument "sentence" must be of type "str"')

    return all(letter in sentence.lower() for letter in string.ascii_lowercase)

    # stillPangram = True

    # for letter in 'abcdefghijklmnopqrstuvwxyz':
    #     if not letter in sentence.lower():
    #         stillPangram = False

    # return stillPangram

File: typescript/pangram/pangram.ts
export function isPangram(sentence: string) {
	// assuming each letter is used only once:

	// if (sentence.length === 0) return false

	// const sanitizedSentence: string = sentence.replace(/ |\.|"/g, '').toLowerCase()

	// const letters = new Set<string>()
	// for (let i: number = 0; i < sanitizedSentence.length; i++) {
	// 	const nextLetter: string = sanitizedSentence[i]
	// 	if (letters.has(nextLetter)) {
	// 		return false
	// 	} else {
	// 		letters.add(nextLetter)
	// 	}
	// }

	// return true

	// assuming each letter can be used more than once:
	const alphabetsLetters: string[] = 'abcdefghijklmnopqrstuvwxyz'.split('')
	const loweredSentence: string = sentence.toLowerCase()
	return alphabetsLetters.every(l => loweredSentence.includes(l))
}
